src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -48,7 +53,11 @@ public String findElement(List<String> elements, String element) {
      * NoSuchElementException</p>
      **/
     public Double averageSumOdd(List<Integer> numbers) {
-        return null;
+        return IntStream.range(0, numbers.size())
+                .mapToDouble(n -> n % 2 == 1 ? numbers.get(n) - 1 : numbers.get(n))
```

Comment:
```suggestion
                .map(n -> n % 2 == 1 ? numbers.get(n) - 1 : numbers.get(n))
```
повинно працювати зі звичайним map